### PR TITLE
Fix for test_voq_ip_fwd

### PR DESCRIPTION
**Pull request by yatishkoul**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
The test was failing due to a Runtime Error caused by modifying the dictionary during iteration.

#### How did you do it?
To resolve this, introduced a separate list to store the keys of the items that need to be removed. After the iteration completes, the keys are removed from the dictionary, preventing the error.
#### How did you verify/test it?
After running this on different duts

### CODE REVIEW

**Review by arlakshm — APPROVED** [review]

**Comment by arlakshm** [comment]
@yatishkoul, can you fix the linter?

**Comment by bingwang-ms** [comment]
> The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request. The pre-commit check is a mandatory check, please fix detected issues.
> 
> Detailed pre-commit check results: trim trailing whitespace.................................................Failed- hook id: trailing-whitespace- exit code: 1- files were modified by this hookFixing tests/voq/test_voq_ipfwd.pyfix end of files.........................................................Passedcheck yaml...........................................(no files to check)Skippedcheck for added large files..............................................Passedcheck python ast.........................................................Passedflake8...................................................................Passedflake8...............................................(no files to check)Skippedcheck conditional mark sort..........................(no files to check)Skipped
> 
> To run the pre-commit checks locally, you can follow below steps:
> 
> 1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
>    the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
>    docker container.
> 2. Ensure that the `pre-commit` package is installed:
> 
> ```
> sudo pip install pre-commit
> ```
> 
> 3. Go to repository root folder
> 4. Install the pre-commit hooks:
> 
> ```
> pre-commit install
> ```
> 
> 5. Use pre-commit to check staged file:
> 
> ```
> pre-commit
> ```
> 
> 6. Alternatively, you can check committed files using:
> 
> ```
> pre-commit run --from-ref <commit_id> --to-ref <commit_id>
> ```

@yatishkoul Can you follow this instruction to install pre-hook locally?

File: tests/common/helpers/sonic_db.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def hget_key_value(self, key, field):
             raise SonicDbKeyNotFound("Key: %s, field: %s not found in sonic-db cmd: %s" % (key, field, cmd))
         else:
             if six.PY2:
-                return result['stdout'].decode('unicode-escape')
+                return result['stdout'].decode('unicode-escape').splitlines()
```
Comment (bingwang-ms) [inline tests/common/helpers/sonic_db.py:93]: Why do we need `splitlines` for Python2 only? It will return a list instead of a string

Comment (yatishkoul) [reply tests/common/helpers/sonic_db.py:93]: it is to support python3 str .decode method issues

File: tests/common/helpers/sonic_db.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def hget_key_value(self, key, field):
...
             else:
                 return result['stdout']
```
[the remark this answers is not in the document]
Comment (yatishkoul) [reply tests/common/helpers/sonic_db.py:95]: in else block, splitlines() would not be required as we are not using decode(), as decode is supported for str in python2 only, so if we had used .decode there, so we would have to use splitlines to make it work for python3.

**Review by SuvarnaMeenakshi — APPROVED** [review]
lgtm

**Pull request merged**